src/DataBuilder.php

```diff
@@ -980,4 +976,33 @@ public function needsTruncating(array $payload)
     {
         return strlen(json_encode($payload)) > self::MAX_PAYLOAD_SIZE;
     }
+
+    /**
+     * Parses an array of code lines from source file with given filename.
+     *
+     * Attempts to automatically detect the line break character used in the file.
+     *
+     * @param string $filename
+     * @return string[] An array of lines of code from the given source file.
+     */
+    private function getSourceLines($filename)
+    {
+        $rawSource = file_get_contents($filename);
+
+        $source = explode(PHP_EOL, $rawSource);
+
+        if (count($source) === 1) {
+            if (substr_count($rawSource, "\n") > substr_count($rawSource, "\r")) {
+                $source = explode("\n", $rawSource);
+                return $source;
+            }
+            
+            $source = explode("\r", $rawSource);
+            return $source;
+        }
+
+        $source = str_replace(array("\n", "\t", "\r"), '', $source);
```

Comment:
Yes. You are right, this is inconsistent (we returned from the if branches without replacing the linebreaks). My mistake. I have fixed it in the next commit.